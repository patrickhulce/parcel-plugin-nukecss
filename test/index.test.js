const path = require('path')
const Bundler = require('parcel-bundler')
const plugin = require('../lib')

const DIST = path.join(__dirname, 'dist')
const simple = path.join(__dirname, 'fixtures/simple.html')

function findCSSBundle(bundle) {
  for (const child of bundle.childBundles) {
    if (child.type === 'css') return child
    const recursive = findCSSBundle(child)
    if (recursive) return recursive
  }
}

/* eslint-env jest */

describe('lib/index.js', () => {
  let bundler, bundle

  describe('simple case', () => {
    beforeAll(async () => {
      bundler = new Bundler(simple, {
        watch: false,
        logLevel: 0,
        production: true,
        outDir: DIST,
      })

      plugin(bundler)
    })

    it('should eliminate unused css', async () => {
      bundle = await bundler.bundle()
      const cssBundle = findCSSBundle(bundle)
      const css = [...cssBundle.assets][0].generated.css

      expect(css).toMatchSnapshot()
    })
  })
})
