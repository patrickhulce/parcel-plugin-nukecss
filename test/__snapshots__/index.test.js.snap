// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/index.js simple case should eliminate unused css 1`] = `".used-html{color:#000}.used-js-key,.used-js-value{color:red}"`;
